specs/sharding/p2p-interface.md

```diff
@@ -66,29 +77,57 @@ Following the same scheme as the [Phase0 gossip topics](../phase0/p2p-interface.
 |----------------------------------|---------------------------|
 | `shard_blob_{shard}`             | `SignedShardBlob`         |
 | `shard_header`                   | `SignedShardHeader`       |
+| `shard_proposer_slashing`        | `ShardProposerSlashing`   |
 
 The [DAS network specification](./das-p2p.md) defines additional topics.
 
 #### Shard blobs: `shard_blob_{shard}`
 
 Shard block data, in the form of a `SignedShardBlob` is published to the `shard_blob_{shard}` subnets.
 
-The following validations MUST pass before forwarding the `signed_blob` (with inner `blob`) on the horizontal subnet or creating samples for it.
+The following validations MUST pass before forwarding the `signed_blob` (with inner `message` as `blob`) on the horizontal subnet or creating samples for it.
 - _[REJECT]_ `blob.shard` MUST match the topic `{shard}` parameter. (And thus within valid shard index range)
 - _[IGNORE]_ The `blob` is not from a future slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance) --
   i.e. validate that `blob.slot <= current_slot`
   (a client MAY queue future blobs for processing at the appropriate slot).
-- _[IGNORE]_ The blob is the first blob with valid signature received for the proposer for the `(slot, shard)` combination.
+- _[IGNORE]_ The `blob` is new enough to be still be processed --
+  i.e. validate that `compute_epoch_at_slot(blob.slot) >= get_previous_epoch(state)`
+- _[IGNORE]_ The blob is the first blob with valid signature received for the `(blob.proposer_index, blob.slot, blob.shard)` combination.
 - _[REJECT]_ As already limited by the SSZ list-limit, it is important the blob is well-formatted and not too large.
 - _[REJECT]_ The `blob.data` MUST NOT contain any point `p >= MODULUS`. Although it is a `uint256`, not the full 256 bit range is valid.
-- _[REJECT]_ The proposer signature, `signed_blob.signature`, is valid with respect to the `proposer_index` pubkey, signed over the SSZ output of `commit_to_data(blob.data)`.
-- _[REJECT]_ The blob is proposed by the expected `proposer_index` for the blob's slot.
+- _[REJECT]_ The proposer signature, `signed_blob.signature`, is valid with respect to the `proposer_index` pubkey.
+- _[REJECT]_ The blob is proposed by the expected `proposer_index` for the blob's slot
+  in the context of the current shuffling (defined by `parent_root`/`slot`).
```

Comment:
This might be an issue. The current ShardBlob does not have a beacon chain reference (I think what you are referring to as `parent_root`) in the definition and so there is no expected shuffling to be verified against.

I believe we *must* have this in the shardblob payload to properly do verifications

If we add it, we might also add `shard_blob_` gossip validation that the `parent_root` or `beacon_root` reference has been processed locally (with the may IGNORE on that condition instead of this condition here that has it implicit

Reply:
It's based on the beacon-block proposer index, forgot about the `parent_root` context. That's a problem indeed, it needs some context for any forking. If we add such context though, shard blobs cannot be replayed anymore on a chain that prunes out some earlier shard blob header inclusions (if we enforce the `parent_root` to be correct on-chain). This may be a good thing though?

Reply:
It references the new `beacon_block_root` field now.